src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIBuildJobExecutionService.java

```diff
@@ -120,6 +160,18 @@ public LocalCIBuildResult runBuildJob(ProgrammingExerciseParticipation participa
         Path assignmentRepositoryPath = assignmentRepositoryUrl.getLocalRepositoryPath(localVCBasePath).toAbsolutePath();
         Path testsRepositoryPath = testsRepositoryUrl.getLocalRepositoryPath(localVCBasePath).toAbsolutePath();
 
+        Path[] auxiliaryRepositoriesPaths;
+
+        if (auxiliaryRepositoriesUrls.length > 0) {
+            auxiliaryRepositoriesPaths = new Path[auxiliaryRepositoriesUrls.length];
+            for (int i = 0; i < auxiliaryRepositoriesUrls.length; i++) {
+                auxiliaryRepositoriesPaths[i] = auxiliaryRepositoriesUrls[i].getLocalRepositoryPath(localVCBasePath).toAbsolutePath();
+            }
+        }
+        else {
+            auxiliaryRepositoriesPaths = new Path[0];
+        }
```

Comment:
Why can't you directly do this when getting the auxiliaryRepositoriesUrls above?